delivery_local_pickup/models/delivery_carrier.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, fields, models
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = "delivery.carrier"
+
+    delivery_type = fields.Selection(selection_add=[("local_pickup", "Local pickup")])
+    local_address_id = fields.Many2one(
+        comodel_name="res.partner", string="Local address"
+    )
+
+    def local_pickup_send_shipping(self, pickings):
+        return [self.local_pickup_create_shipping(p) for p in pickings]
+
+    def local_pickup_create_shipping(self, picking):
+        return {"exact_price": False, "tracking_number": _("Local pickup")}
```

Comment:
Testing on runbot the other day, the mail stated something like "your trackink numbre is Local pickup...". Maybe it would be more coherent to simply put the picking name.

Reply:
I think it is important that this text appears because it is what is shown in the email that is sent to the client when the stock picking is done.

Reply:
Ok, but the resultant message is kinda weird. Anyway I don't come up with any clever string substitute right now :slightly_smiling_face: 

![Selección_01064](https://user-images.githubusercontent.com/5040182/133968162-67b36342-8bea-4ab6-b850-2ceb5deafff2.png)
:

Reply:
Ok yeah, it's kind of weird but there is no other option in my opinion.
Can you check the PR again to approve it if it is OK?

Reply:
I'd go for a picking name plus carrier name. For example, for the picking *WH/OUT/0001* and the carrier name *Store Pickup* we would get something like: "*Your tracking reference is [**WH/OUT/0001 (Store Pickup)**](#)*"

Which way more informative, don't you think?

Reply:
Thanks, changes done.